# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/multi-value/call-validate.js
# Commit: e71e54d93af8
# Full Hash: e71e54d93af8baa6ef915c81dc04ac3400f825cd
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-06 20:10:41
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r afbf73fb4073 -r e71e54d93af8 js/src/jit-test/tests/wasm/multi-value/call-validate.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/wasm/multi-value/call-validate.js	Fri Mar 06 13:53:07 2020 +0000
@@ -0,0 +1,47 @@
+wasmValidateText(`
+  (module
+    (func (result i32 i32)
+      (i32.const 32)
+      (i32.const 10)))`);
+
+wasmValidateText(`
+  (module
+    (type $t (func (result i32 i32)))
+    (func (type $t)
+      (i32.const 32)
+      (i32.const 10)))`);
+
+wasmValidateText(`
+  (module
+    (func (result i32 i32)
+      (block (result i32 i32)
+        (i32.const 32)
+        (i32.const 10))))`);
+
+wasmValidateText(`
+  (module
+    (func $return-2 (result i32 i32)
+      (i32.const 32)
+      (i32.const 10))
+    (func $tail-call (result i32 i32)
+      (call 0)))`);
+
+wasmValidateText(`
+  (module
+    (func $return-2 (result i32 i32)
+      (i32.const 32)
+      (i32.const 10))
+    (func $add (result i32)
+      (call 0)
+      i32.add))`);
+
+wasmValidateText(`
+  (module
+    (func $return-2 (param i32 i32) (result i32 i32)
+      (local.get 0)
+      (local.get 1))
+    (func (export "run") (result i32)
+      (i32.const 32)
+      (i32.const 10)
+      (call 0)
+      i32.add))`);